### PR TITLE
Extend leap analyzer with optimal solution

### PR DESCRIPTION
**Pull request by EarthlingRich**
This solution (https://exercism.io/mentor/analyses/1912) is referred to a mentor while it could be seen as a optimal solution. This PR add this solution to the leap analyzer as a optimal solution.

### CODE REVIEW

File: src/Exercism.Analyzers.CSharp/Analyzers/Leap/LeapSyntaxFactory.cs [hunk 1]
```diff
@@ -22,6 +22,19 @@ internal static class LeapSyntaxFactory
                 LeapModuloExpression(leapSolution, 400),
                 NumericLiteralExpression(0)));
 
+        public static BinaryExpressionSyntax LeapMinimumNumberOfChecksWithoutParenthesesBinaryExpressionReversed(LeapSolution leapSolution) =>
+            LogicalOrExpression(
+                EqualsExpression(
```
Comment (ErikSchierboom) [inline src/Exercism.Analyzers.CSharp/Analyzers/Leap/LeapSyntaxFactory.cs]: The `EqualsExpression` is shared with the method above. Could you extract it to a helper method?

File: src/Exercism.Analyzers.CSharp/Analyzers/Leap/LeapSyntaxFactory.cs [hunk 2]
```diff
@@ -22,6 +22,19 @@ internal static class LeapSyntaxFactory
...
+                    LeapModuloExpression(leapSolution, 400),
+                    NumericLiteralExpression(0)),
+            LogicalAndExpression(
+                NotEqualsExpression(
```
Comment (ErikSchierboom) [inline src/Exercism.Analyzers.CSharp/Analyzers/Leap/LeapSyntaxFactory.cs]: The `NotEqualsExpression` is shared with the method above. Could you extract it to a helper method?

File: src/Exercism.Analyzers.CSharp/Analyzers/Leap/LeapSyntaxFactory.cs [hunk 3]
```diff
@@ -22,6 +22,19 @@ internal static class LeapSyntaxFactory
...
+                    LeapModuloExpression(leapSolution, 100),
+                    NumericLiteralExpression(0)),
+            EqualsExpression(
```
Comment (ErikSchierboom) [inline src/Exercism.Analyzers.CSharp/Analyzers/Leap/LeapSyntaxFactory.cs]: The `EqualsExpression` is shared with the method above. Could you extract it to a helper method?

**Comment by EarthlingRich** [comment]
Good point about the helper methods, I added them. Thanks!

**Pull request merged**